[Ben926] iP

Ben926 opened the pull request:
### Ronaldo: Your Ultimate Task Manager🔥

> "The best Task Manager I've ever used!" John Doe

Features 

- Text-based
- Intuitive
- Easy to use!
- ~~Fast~~ SUPER fast to use!

Simply
1. Download it from [here](https://github.com/Ben926/ip)
2. Open it
3. Start adding and managing tasks!

And its **free**!

Task Types:
- [ ] Todos
- [ ] Deadlines
- [ ] Events

Here's a little sneak peek into some of our carefully written code!
```Java
public void run() {
        ui = new Ui();
        Storage storage = new Storage(filePath);
        tasks = new TaskList();

        ui.printWelcomeText();
        while (isRunning) {
            try {
                String line = ui.readCommand();
                Command command = Parser.parseCommand(line);
                isRunning = this.executeCommand(command, line);
            } catch (RonaldoException e) {
                System.out.println(e.getMessage());
            } finally {
                storage.saveTasks(tasks);
            }
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Ben926 committed:
rename, greet, exit

Ben926 committed:
Echo

Ben926 committed:
add, list

Ben926 committed:
mark as done

Ben926 committed:
added diff class files for task

Ben926 committed:
added support for diff tasks input

Ben926 committed:
added automated testing in text-ui-test files

Ben926 committed:
added exception handling, cleanup

Ben926 committed:
added delete functionality

Ben926 committed:
added enums for command

Ben926 committed:
Add saveTasks method

Ben926 committed:
Update output format

Ben926 committed:
Merge branch 'branch-Level-7'

Ben926 committed:
Add datetime handling for deadline tasks

Ben926 committed:
Add datetime handling for event tasks

Ben926 committed:
Modify string representation of deadline and event tasks

Ben926 committed:
Merge branch 'branch-Level-8'

Ben926 committed:
Add new class files with skeleton features

Ben926 committed:
Add run logic in Ronaldo.java and more features in new class files

Ben926 committed:
Add full OOP logic without changing main entry method yet

Ben926 committed:
Rewrite main program to use new classes

Ben926 committed:
Merge branch 'branch-A-MoreOOP'

Ben926 committed:
Add java files into ronaldo package

Ben926 committed:
Merge branch 'branch-A-packages'

Ben926 committed:
Merge branch 'add-gradle-support'

Ben926 committed:
Add ParserJUnit test

Ben926 committed:
Add Storage JUnit test

Ben926 committed:
Merge branch 'branch-A-JUnit'

Ben926 committed:
Add MANIFEST.MF

Ben926 committed:
Add JavaDoc comments to all non-private classes and methods

Ben926 committed:
Merge branch 'branch-A-JavaDoc'

Ben926 committed:
Tweak styling to follow convention

Ben926 committed:
Merge branch 'branch-A-CodingStandard'

Ben926 committed:
Add Find command and processing of command

Ben926 committed:
Fix merge conflicts

Ben926 committed:
Add GUI with JavaFX

Ben926 committed:
Fix minor coding standard violations

Ben926 committed:
Merge branch 'branch-Level-10'

Ben926 committed:
Remove scanner

Zhannyhong commented on src/main/java/ronaldo/Task.java:
```diff
@@ -0,0 +1,60 @@
+
+    public boolean checkMatchingDescription(String subDescription) {
+        return this.description.contains(subDescription);
+    }
```
Consider changing the function name to sound like a boolean

Zhannyhong commented on src/main/java/ronaldo/Parser.java:
```diff
@@ -0,0 +1,139 @@
+                    + "Try typing one of the following: \nhello\nlist\ntodo <desc>\ndeadline <desc> /by <date>\n"
+                    + "event <desc> /from <start> /to <end>\nmark <num>\nunmark <num>\ndelete <num>\nfind <keyword>\nbye (to exit the program)\n");
+
+        }
```
Since there are many if/else blocks, perhaps a switch case can be used instead to compare the first word?

Zhannyhong commented on src/main/java/ronaldo/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     */
+    public static Task parseEventCommand(String line) throws RonaldoException {
+        try {
+            String[] tokens = line.split("/from|/to");
```
What will happen if the user specifies 2 /from options instead? How can you ensure that the arguments are in the correct format?

Zhannyhong commented on src/main/java/ronaldo/Parser.java:
```diff
@@ -0,0 +1,139 @@
+        if (index > maxIndex) {
+            throw new RonaldoException(String.format("Don't be stupid. No such task exists! " +
+                    "The list only has %d items.\n", maxIndex + 1));
+        }
```
What if a negative integer was supplied?

Zhannyhong commented on src/main/java/ronaldo/Task.java:
```diff
@@ -0,0 +1,60 @@
+     * @return A string representing the task's status icon.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
```
The parentheses is unnecessary

Zhannyhong commented on src/main/java/ronaldo/TaskList.java:
```diff
@@ -0,0 +1,97 @@
+
+    public TaskList findMatchingTasks(String subDescription) {
+        return new TaskList(new ArrayList<Task>(this.arr.stream()
+                .filter(task -> task.checkMatchingDescription(subDescription)).toList()));
```
Perhaps can just convert the stream directly into an array using the .toArray method

Zhannyhong commented on src/main/java/ronaldo/Ui.java:
```diff
@@ -0,0 +1,98 @@
+     * @return The welcome message as a string.
+     */
+    public String getWelcomeText() {
+        return "SIIUUUUUU!!! Ronaldo here.\nHow can the GOAT help you?\n";
```
SIUUUUUUUU 🙅🏽 🧍🏽

Zhannyhong reviewed (commented):
LGTM. Some nits to pick

Judy1x4 commented on src/main/java/ronaldo/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A Task with a single due date
```
Perhaps its good to start with verb with s at the end? e.g. "Represents... "

Judy1x4 commented on src/test/java/ronaldo/StorageTest.java:
```diff
@@ -0,0 +1,60 @@
+import java.nio.file.Files;
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Shouldn't all the imports be explicitly stated?

Judy1x4 commented on src/main/java/ronaldo/MainWindow.java:
```diff
@@ -0,0 +1,58 @@
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
```
After the first sentence, shouldn't the following sentences be a full sentence instead of starting with an action verb?

Judy1x4 commented on src/main/java/ronaldo/Task.java:
```diff
@@ -0,0 +1,60 @@
+        this.isDone = false;
+    }
+
+    public boolean checkMatchingDescription(String subDescription) {
```
Perhaps the function name sound more like a boolean?

Judy1x4 commented on src/main/java/ronaldo/Storage.java:
```diff
@@ -0,0 +1,46 @@
+     *
+     * @param textFilePath The path to the file where the text file containing the tasks will be stored.
+     */
+    public Storage (String textFilePath) {
```
Shouldn't the constructor name stays attached to the open parenthesis ( ?

Judy1x4 commented on src/main/java/ronaldo/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    @Override
+    public String toString() {
+        return "[D] " + super.toString() + String.format(" (by: %s)",
```
Perhaps preferrably the higher-level breaks should be applied instead of lower-level breaks. Perhaps consider a line break right after the open paranthesis of String.format?

Judy1x4 reviewed (commented):
Really a clean and neat code! LGTM! There are some violation of coding standards to be adjusted.

Ben926 committed:
Add ronaldo.txt to gitignore

Ben926 committed:
Add assertions to enforce code assumptions

Added assertions in key methods to ensure critical assumptions hold
true during runtime. This improves code robustness by catching
unexpected states early.

- Ensured  is not null or empty in the  constructor.
- Verified  is non-null and non-empty in the  method.
- Checked  is initialized before performing operations.
- Prevented marking, unmarking, or deleting tasks from an empty list.
- Ensured  updates correctly after adding new tasks.

These assertions help maintain code correctness and prevent silent
failures, making debugging easier.

Ben926 committed:
Merge pull request #1 from Ben926/branch-A-Assertions

Add assertions to enforce code assumptions

Ben926 committed:
Improve code quality and consistency across the project

Refactored various parts of the codebase to enhance readability,
consistency, and maintainability. This includes style improvements,
better method naming, documentation enhancements, and import
organization.

- Updated JavaDoc comments for clarity and consistency.
- Ensured all import statements are explicitly stated.
- Reformatted constructor definition style ( class).
- Renamed  to a more boolean-like method name.
- Adjusted method documentation to use complete sentences where needed.
- Ensured class-level documentation follows proper structure.
- Standardized  methods for improved output formatting.
- Improved JavaDoc formatting to maintain consistency across classes.

These changes improve the readability and maintainability of the code, making future modifications easier and reducing ambiguity in method
documentation.

Ben926 committed:
Merge branch 'master' of https://github.com/Ben926/ip into branch-A-CodeQuality

Ben926 committed:
Merge pull request #2 from Ben926/branch-A-CodeQuality

Branch a code quality

Ben926 committed:
Add sort command

Ben926 committed:
Merge branch 'branch-C-Sort'

Ben926 committed:
edit Readme

Ben926 committed:
Merge branch 'A-UserGuide'

Ben926 committed:
Edit docs/readme.md

Ben926 committed:
Edit javadoc comments

Ben926 committed:
Edit README.md

Ben926 committed:
Edit Parser.java

Ben926 committed:
Edit Task.java

Ben926 committed:
Edit UI.java

Ben926 committed:
Edit README.md

Ben926 committed:
Fix javafx components missing error
